### ISSUE TITLE
Introduction of an OxyException type

### ISSUE THREAD

**Issue by maettu-this** [oxyplot/oxyplot#1464]
### Feature description

OxyPlot throws of type `Exception` at approx. 25 locations. Proposing to throw of type `OxyException` instead, allowing users to implement dedicated exception handling per exception type.

**Comment by objorke** [comment]
Is this according to the best practices? https://docs.microsoft.com/en-us/dotnet/standard/exceptions/best-practices-for-exceptions

**Comment by objorke** [comment]
Search for "throw new Exception" - I think these should be replaced by more detailed or custom exception types.

**Comment by objorke** [comment]
I agree it is a good idea to review all the exceptions to make sure it is consistent.

**Comment by maettu-this** [comment]
I haven't found a note on that best practise in the above mentioned link, but https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1031 requests to not catch the general `Exception` type, i.e. emphasizes to always use a to more specific exception type. Also, I don't know of any framework class throwing a general exception type.

**Comment by VisualMelon** [comment]
There are quite a number that should be `ArgumentException` or variations on that theme. I'll put together a PR replacing those, and try to make a note of the sort of circumstances where `Exception`s are being thrown otherwise.

**Comment by VisualMelon** [comment]
Most of the `Exception`s appear in imported code for compression. Some of these are addressed in e85a0cf. Some changes in e85a0cf may be contentious (some tests have been updated), but mostly it is providing more explicit argument exceptions, normalising the message formatting, and using `nameof` for parameter names.

I think that changing the remaining `Exception`s to `InvalidOperationException`s would be consistent with much of the rest of the library.

I've made some [rough notes](https://gist.github.com/VisualMelon/f5fe19409609610417dc32c14740443e) on other exceptions in the core library. Switches over enums which can't handle particular/default cases, for example, are somewhat inconsistent, but generally the pattern is to throw an `ArgumentOutOfRange` exception when applicable, but this is not completely consistent.

**Comment by Avinash-Sahani** [comment]
@maettu-this  is it open for contribution

**Comment by maettu-this** [comment]
> [@maettu-this](https://github.com/maettu-this) is it open for contribution

Whether a solution to this issue accepts a contribution? This has to be decided by the development team (@objorke, @VisualMelon). But given the labels [enhancement](https://github.com/oxyplot/oxyplot/issues?q=state%3Aopen%20label%3A%22enhancement%22) and [good first issue](https://github.com/oxyplot/oxyplot/issues?q=state%3Aopen%20label%3A%22good%20first%20issue%22) I assume so.
